### PR TITLE
Implement check for pointer to sync.Mutex

### PR DESCRIPTION
**Pull request by mirshahriar**
resolve issue: #2

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,6 @@
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
+.idea
```
Comment (sadlil) [inline .gitignore]: ignore the whole folder? use `.idea/` `bin/`

File: analyzers/mutexpointer/mutex_pointer.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+		}
+		i.pass.Reportf(expr.Pos(), "mutex-pointer: var %s uses pointer to mutex", varName)
+
+		ret = true
+	}
+
+	return ret
+}
+
+func check(node ast.Expr) bool {
```
Comment (sadlil) [inline analyzers/mutexpointer/mutex_pointer.go]: move this function to down below?

File: analyzers/mutexpointer/testdata/src/testpkg/code.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package pkg
```
Comment (sadlil) [inline analyzers/mutexpointer/testdata/src/testpkg/code.go]: call `pkg` to `testpkg`?

File: analyzers/mutexpointer/testdata/src/testpkg/code.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	invalidMutextInBlock *sync.Mutex // want "mutex-pointer: var invalidMutxtInBlock uses pointer to sync.Mutex"
+	validMutex            sync.Mutex
+)
+
+type rwm *sync.RWMutex // want "mutex-pointer: var rwm points to pointer of sync.Mutex"
+
+func testFunc() {
+	var anotherMutexPointer *sync.Mutex // want "mutex-pointer: var anotherMutexPointer uses pointer to sync.Mutex"
+	_ = anotherMutexPointer
+	alsoInvalidMutex, valid, notValidToo := new(sync.Mutex), sync.Mutex{}, new(sync.RWMutex)
```
Comment (sadlil) [inline analyzers/mutexpointer/testdata/src/testpkg/code.go]: Keep a TODO if `new(sync.Mutex)` is not valid, so that we can add implementations later.

**Review by sadlil — CHANGES_REQUESTED** [review]

File: analyzers/mutexpointer/testdata/src/testpkg/code.go (hunk 4 above)
Comment (mirshahriar) [reply analyzers/mutexpointer/testdata/src/testpkg/code.go]: `new(sync.Mutex)` is already checked

Comment (sadlil) [reply analyzers/mutexpointer/testdata/src/testpkg/code.go]: i don't see any want document that validates the check?

**Comment by sadlil** [comment]
Also can please update the Readme to include what analyzers are currently supported. Thank you.

File: analyzers/mutexpointer/testdata/src/testpkg/code.go (hunk 4 above)
Comment (mirshahriar) [reply analyzers/mutexpointer/testdata/src/testpkg/code.go]: added

**Review by sadlil — APPROVED** [review]

**Pull request merged**